lib/db/task.ts

```diff
@@ -0,0 +1,114 @@
+import { start } from "repl";
+import prisma from "./prisma";
+import { Task } from "@prisma/client";
+
+/**
+ * Retrieves a single task based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Partial<Task>} params Information about the task.
+ * @returns {Promise<Task | null>}
+ *} The retrieved task object.
+ */
+export async function getTask(params: Partial<Task>): Promise<Task | null> {
+  const task = await prisma.task.findFirst({
+    where: {
+      ...params,
+    },
+    include: {
+      user: true,
+      activities: true,
+      comments: true,
+      status: true,
+      sprint: true,
+    },
+  });
+  return task;
+}
+
+/**
+ * Retrieves multiple tasks based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Partial<Task>} params Information about the tasks to retrieve.
+ * @returns {Promise<Task[]>} An array of retrieved task objects.
+ */
+export async function getTasks(params: Partial<Task>): Promise<Task[]> {
+  const task = await prisma.task.findMany({
+    where: {
+      ...params,
+    },
+    include: {
+      user: true,
+      activities: true,
+      comments: true,
+      status: true,
+      sprint: true,
+    },
+  });
+  return task;
+}
+
+/**
+ * Creates a new task.
+ *
+ * @export
+ * @async
+ * @param {Partial<Task>} params Information about the task to create.
+ * @returns {Promise<Task>} The created task object.
+ */
+export async function createTask(params: any): Promise<Task> {
+  const task = await prisma.task.create({
+    data: {
+      title: params.title,
+      description: params.description,
+      category: params.category,
+      requirements: params.requirements,
+      acceptance_criteria: params.acceptance_criteria,
+      points: params.points,
+      assigned_to: params.assigned_to,
+      sprint: { connect: { id: params.sprint_id } },
+      status: { connect: { id: params.status_id } },
+      priority: { connect: { id: params.priority_id } },
+    },
+  });
+  return task;
+}
+
+/**
+ * Updates an existing task.
+ *
+ * @export
+ * @async
+ * @param {Partial<Task>} params Information about the task to update.
+ * @returns {Promise<Task>} The updated task object.
+ */
+export async function updateTask(params: Partial<Task>): Promise<Task> {
+  const task = await prisma.task.update({
```

Comment:
We may want to use the custom types for the `update` functions as well...

Additionally, if the spread syntax isn't flexible for the mapped relations, then we might have to create additional functions (such as `moveToNewSprint` or `updateTaskPriority` in a future story...

Reply:
@winstuhn

Reply:
Updated all return types to include the custom types.